src/coreclr/jit/lower.cpp

```diff
@@ -2591,6 +2591,25 @@ GenTree* Lowering::OptimizeConstCompare(GenTree* cmp)
 
         if (op2Value != 0)
         {
+            // Optimizes (X & 1) == 1 to (X & 1)
+            // The compiler requires jumps to have relop operands, so we do not fold that case.
+            LIR::Use cmpUse;
+            if ((op2Value == 1) && BlockRange().TryGetUse(cmp, &cmpUse))
```

Comment:
This still needs a check for `cmp->OperIs(GT_EQ)`.

Reply:
I added the checks and moved the expensive operation as you had suggested. These changes give much better diffs, there are not regressions present. However there is something wrong with the tests on arm64 osx systems. I tried to build and run the clr tests on m1 osx laptop and It didn't give me anything suspicious.

Reply:
I think the failures are not related at this point; we're in the middle of a period of some CI instability right now.